src/test/java/be/syntra/devshop/DevshopBack/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,81 @@
+package be.syntra.devshop.DevshopBack.controllers;
+
+import be.syntra.devshop.DevshopBack.entities.User;
+import be.syntra.devshop.DevshopBack.models.UserDto;
+import be.syntra.devshop.DevshopBack.services.UserServiceImpl;
+import be.syntra.devshop.DevshopBack.testutilities.JsonUtils;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.ResultActions;
+
+import java.util.List;
+
+import static be.syntra.devshop.DevshopBack.testutilities.UserUtils.createUserDto;
+import static be.syntra.devshop.DevshopBack.testutilities.UserUtils.createUserList;
+import static org.mockito.Mockito.*;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@Import(JsonUtils.class)
+@WebMvcTest(UserController.class)
+public class UserControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private JsonUtils jsonUtils;
+
+    @MockBean
+    private UserServiceImpl userService;
+
+
+    @Test
+    void createUserEndPointsTest() throws Exception {
+        // given
+        UserDto userDtoDummy = createUserDto();
+        // when
+        ResultActions resultActions =
+                mockMvc.perform(
+                        post("/users")
+                                .contentType(APPLICATION_JSON)
+                                .content(jsonUtils.asJsonString(userDtoDummy))
+                );
+        // then
+        resultActions
+                .andExpect(status().isCreated())
+                .andExpect(content().contentType(APPLICATION_JSON));
+
+
+        verify(userService, times(1)).save(userDtoDummy);
+
+    }
+
+    @Test
+    void testRetrieveAllUserEndpoint() throws Exception {
+        // given
+        List<User> userList = createUserList();
+        when(userService.findAll()).thenReturn(userList);
+        // when
+        MvcResult response =
+                mockMvc.perform(
+                        get("/users").accept(MediaType.APPLICATION_JSON))
+                        .andExpect(status().isOk())
+                        .andReturn();
+
+
+        // then
```

Comment:
then: here you will check if the 'reponse' is ok by using:
`response.andExpect(...)`
also: in the this test's given it says to return a userList when service.findAll() is called, the /get will do that so you should check if its in the response here, if there's data send back after the call, you should check the data, pls look at **testRetrieveAllProductsEndpoint**